test/framework/clusterctl/client.go

```diff
@@ -219,16 +219,19 @@ func UpgradeWithBinary(ctx context.Context, binary string, input UpgradeInput) {
 
 	cmd := exec.Command(binary, args...) //nolint:gosec // We don't care about command injection here.
 
-	out, err := cmd.CombinedOutput()
-	_ = os.WriteFile(filepath.Join(input.LogFolder, "clusterctl-upgrade.log"), out, 0644) //nolint:gosec // this is a log file to be shared via prow artifacts
-	var stdErr string
-	if err != nil {
-		var exitErr *exec.ExitError
-		if errors.As(err, &exitErr) {
-			stdErr = string(exitErr.Stderr)
+	Eventually(func() error {
```

Comment:
let's use a for loop and make the number of retries dependent on the clusterctl version used (>= v1.7.0 should not do any retries)